File: sdk/platform/Web/CalibrationHelpers.js
posemeshModule.CalibrationHelpers = null;

__internalPosemeshAPI.builderFunctions.push(function() {
    __internalPosemesh.CalibrationHelpers.getCalibrationMatrix = function(domain, observed, onlyRotateAroundY) {
        return __internalPosemesh.CalibrationHelpers.__getCalibrationMatrix(domain, observed, onlyRotateAroundY);
    };

    posemeshModule.CalibrationHelpers = __internalPosemesh.CalibrationHelpers;
});

File: sdk/platform/Web/src/CalibrationHelpers.cpp
#include <Posemesh/CalibrationHelpers.hpp>
#include <cassert>
#include <emscripten/bind.h>
#include <memory>

using namespace emscripten;
using namespace psm;

namespace {
std::shared_ptr<psm::Matrix4x4> getCalibrationMatrix(const Pose& domain, const Pose& observed, bool onlyRotateAroundY)
{
    return std::make_shared<psm::Matrix4x4>(CalibrationHelpers::getCalibrationMatrix(domain, observed, onlyRotateAroundY));
}
}

EMSCRIPTEN_BINDINGS(CalibrationHelpers)
{
    class_<CalibrationHelpers>("CalibrationHelpers")
        .class_function("__getCalibrationMatrix(domain, observed, onlyRotateAroundY)", &getCalibrationMatrix, nonnull<ret_val>());
}
